uqcsbot/scripts/leet.py

```diff
@@ -0,0 +1,167 @@
+from uqcsbot import bot, Command
+from http import HTTPStatus
+from uqcsbot.utils.command_utils import loading_status, UsageSyntaxException
+import json
+import requests
+import random
+from slackblocks import Attachment, SectionBlock
+
+
+LC_DIFFICULTY_MAP = ["easy", "medium", "hard"] # leetcode difficulty is 1,2,3, need to map
+HR_DS_API_LINK = ("https://www.hackerrank.com/rest/contests/master/tracks/" +
+    "data-structures/challenges?limit=200")
+HR_ALG_API_LINK = "https://www.hackerrank.com/rest/contests/master/tracks/algorithms/challenges?limit=200"
+LC_API_LINK = 'https://leetcode.com/api/problems/all/'
+
+EASY_COLOR_CODE = "#5db85b"
+MED_COLOR_CODE = "#f1ad4e"
+HARD_COLOR_CODE = "#d9534f"
+
+@bot.on_command('leet')
+@loading_status
+def handle_leet(command: Command) -> None:
+    """
+    `!leet [`easy` | `medium` | `hard`] - Retrieves a set of questions from online coding
+    websites, and posts in channel with a random question from this set. If a difficulty
+    is provided as an argument, the random question will be restricted to this level of 
+    challenge. Else, a random difficulty is generated to choose.
+    """
+
+    difficulty = ""
```

Comment:
Prefer `None` and checking for presence with `is None`, but realistically this is unnecessary

Reply:
removed.

Reply:
resolved.